### PR TITLE
Removing logging exception through update run metadata

### PR DESCRIPTION
**Pull request by jjanezhang**
We are moving to catching exceptions through the python excepthook override 7/8/24. We can remove logging to metadata then.

### CODE REVIEW

File: scripts/train/train.py [hunk 1]
```diff
@@ -427,9 +425,7 @@ def main(cfg: DictConfig) -> Trainer:
             if eval_gauntlet_callback is not None:
                 callbacks.append(eval_gauntlet_callback)
         except BaseContextualError as e:
-            if mosaicml_logger is not None:
-                e.location = EvalDataLoaderLocation
-                mosaicml_logger.log_exception(e)
+            e.location = EvalDataLoaderLocation
```
Comment (mvpatel2000) [inline scripts/train/train.py:428]: why does location need to be set if the logger is being migrated off of? should we be removing this fn from logger in composer?

Comment (jjanezhang) [reply scripts/train/train.py:428]: The location attribute is on the exception class helps us tell the user if their data validation failed for their eval dataset or their train dataset. We still need this attribute set even when we are catching exceptions without the logger.

**Review by mvpatel2000 — APPROVED** [review]
LGTM

**Review by mvpatel2000 — APPROVED** [review]
LGTM

**Comment by jjanezhang** [comment]
> Member

removed the logger from a couple more places, requested review again

File: scripts/data_prep/convert_delta_to_json.py [hunk 2]
```diff
@@ -667,18 +663,10 @@ def fetch_DT(args: Namespace) -> None:
...
-
-    try:
-        w = WorkspaceClient()
-        args.DATABRICKS_HOST = w.config.host
-        args.DATABRICKS_TOKEN = w.config.token
-
-        tik = time.time()
-        fetch_DT(args)
-        log.info(f'Elapsed time {time.time() - tik}')
+    w = WorkspaceClient()
```
Comment (mvpatel2000) [inline scripts/data_prep/convert_delta_to_json.py:666]: does this import need to be gated?

**Review by mvpatel2000 — APPROVED** [review]

File: llmfoundry/callbacks/run_timeout_callback.py [hunk 3]
```diff
@@ -8,19 +8,18 @@
...
 
-def _timeout(timeout: int, mosaicml_logger: Optional[MosaicMLLogger] = None):
+def _timeout(timeout: int):
     log.error(f'Timeout after {timeout} seconds of inactivity after fit_end.',)
-    if mosaicml_logger is not None and no_override_excepthook():
-        mosaicml_logger.log_exception(RunTimeoutError(timeout=timeout))
-    os.kill(os.getpid(), signal.SIGINT)
+    try:
+        raise RunTimeoutError(timeout=timeout)
+    except RunTimeoutError:
```
Comment (irenedea) [inline llmfoundry/callbacks/run_timeout_callback.py:21]: this seems like a noop? Can we remove this?

**Review by irenedea — APPROVED** [review]
LGTM! minor comment

File: scripts/data_prep/convert_delta_to_json.py (hunk 2 above)
Comment (jjanezhang) [reply scripts/data_prep/convert_delta_to_json.py:666]: I don't think so, we aren't currently in gating it in the script.

**Pull request merged**